brian2/devices/cpp_standalone/device.py

```diff
@@ -906,14 +907,45 @@ def write_static_arrays(self, directory):
             arr.tofile(os.path.join(directory, 'static_arrays', name))
             static_array_specs.append((name, c_data_type(arr.dtype), arr.size, name))
         self.static_array_specs = static_array_specs
-            
+    
+    def get_all_objects(self, net):
+        all_objects = []
+        def add_child_objects(*objs):
+            for obj in objs:           
+                if isinstance(obj, BrianObject):
+                    if obj not in all_objects:  # Don't include objects twice
+                        all_objects.append(obj)
+                    add_child_objects(obj.contained_objects)
+                else:
+                    # allow adding values from dictionaries
+                    if isinstance(obj, Mapping):
+                        add_child_objects(*obj.values())
+                    else:
+                        try:
+                            for o in obj:
+                                if o is obj:
+                                    raise TypeError()
+                                add_child_objects(o)
+                        except TypeError:
+                            raise TypeError("Can only add objects of type BrianObject, "
+                                            "or containers of such objects to Network")
+        add_child_objects(net.objects)
+        when_to_int = dict((when, 1+i*3)
+                           for i, when in enumerate(net.schedule))
+        when_to_int.update(('before_' + when, i*3)
+                           for i, when in enumerate(net.schedule))
+        when_to_int.update(('after_' + when, 2+i*3)
+                           for i, when in enumerate(net.schedule))
+        all_objects.sort(key=lambda obj: (when_to_int[obj.when],obj.order,obj.name))
+        return(all_objects)
+        
     def find_synapses(self):
         # Write the global objects
         networks = [net() for net in Network.__instances__()
                     if net().name != '_fake_network']
         synapses = []
         for net in networks:
-            net_synapses = [s for s in net.objects if isinstance(s, Synapses)]
+            net_synapses = [s for s in self.get_all_objects(net) if isinstance(s, Synapses)]
```

Comment:
`Synapses` are never `contained_objects` of other objects, so looking at `net.objects` should be enough here.